core.osgi/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/osgi/commands/RepairStatusCommand.java

```diff
@@ -0,0 +1,289 @@
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.core.osgi.commands;
+
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.ericsson.bss.cassandra.ecchronos.core.Clock;
+import com.ericsson.bss.cassandra.ecchronos.core.metrics.TableRepairMetricsProvider;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.RepairConfiguration;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.RepairJobView;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.RepairScheduler;
+import com.ericsson.bss.cassandra.ecchronos.core.utils.TableReference;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.support.ansi.SimpleAnsi;
+import org.apache.karaf.shell.support.table.ShellTable;
+
+@Service
+@Command(scope = "repair", name = "status", description = "Give the current repair status")
+public class RepairStatusCommand implements Action
+{
+    @Option(name = "-l", aliases = "--limit", description = "Number of entries to display", required = false, multiValued = false)
+    int limit = Integer.MAX_VALUE;
+
+    @Option(name = "-s", aliases = "--sort_by", description = "Sort output based on TABLE_NAME/STATUS/REPAIRED_RATIO/REPAIRED_AT/NEXT_REPAIR", required = false, multiValued = false)
+    SortBy sortBy = SortBy.TABLE_NAME;
+
+    @Option(name = "-r", aliases = "--reverse", description = "Reverse the sort order", required = false, multiValued = false)
+    boolean reverse = false;
+
+    @Option(name = "-a", aliases = "--all", description = "Show status for all tables (including complete status)", required = false, multiValued = false)
+    boolean showAll = false;
+
+    @Option(name = "--summary", description = "Show summary only", required = false, multiValued = false)
+    boolean summaryOnly = false;
+
+    @Option(name = "--no-format", description = "Disable output formatting of table and colors", required = false, multiValued = false)
+    boolean noFormat = false;
+
+    @Reference
+    private RepairScheduler myRepairScheduler;
+
+    @Reference
+    private TableRepairMetricsProvider myTableRepairMetrics;
+
+    private Clock myClock = Clock.DEFAULT;
```

Comment:
Might be worth trying to switch from the custom implementation to the Clock in Java8 https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html